### DISCUSSION TITLE
Protein Inference type & Heuristics setting

### DISCUSSION THREAD

**Discussion by JM-Bader** [vdemichev/DiaNN#458] [Q&A]
DIANN offers a dropdown Protein inference (settings: Off, Genes, Genes species-specific, Protein names from FASTA, Isoform IDs). 
Second, protein inference can be done in a heuristic way by ticking the button.

Q1: Is protein inference in DIANN identical with protein grouping (i.e. assignment of identified peptides to protein groups) ? If not, what is the difference?
Q2: heuristic protein inference coerces aggregation at the specified higher level selected and thus prevents that a single proteoform occurs in multiple entries with different quantification due to varying levels of razor peptide inclusion. Correct?
Q3: the dropdown menu 'Protein inference' specifies that higher level at which protein grouping is performed. Correct?
Q4: Accordingly, the heuristic setting and setting of the aggregation level 'Protein inference' are independent and all combinations are legitimate. Correct? E.g. heuristic grouping at Isoform ID level ? 

Q5: Does heuristic protein inference have an effect, when protein inference is set to off (when using a project-specific library)?
Q6: What happens if a project-specific library with UniProt IDs is used (i.e. proteins information present) but there is protein grouping done (as is typical for some libraries) and one turns protein inference (I) off vs (II) on -no heuristic vs (III) on +heuristic ?

**Comment by vdemichev** [comment]
Q1: Yes, the term 'protein inference' refers to protein grouping.
Q2: Yes.
Q3: No, it specifies the definition of proteotypicity and gives a 'hint' to the protein grouping algorithm which helps to choose the most appropriate protein for each precursor. The effect of this setting on protein grouping is really minor.
Q4: Yes.
Q5: No.
Q6: If the library has already grouped proteins in the UniprotIDs column, then DIA-NN's protein grouping will have minor effect if any, when turned on. But it does not make sense to turn it on in this case.

**Reply by JM-Bader** [reply]
thanks!
Re Q3: It appears that I get drastically fewer quantified proteins in a library-free search when using Protein inference set to Isoform IDs compared to Protein inference set to Genes. 

Isoform IDs: 
![image](https://user-images.githubusercontent.com/98888664/183425914-10ccfedb-48b0-4147-95de-c5178764908b.png)

Genes: 
![image](https://user-images.githubusercontent.com/98888664/183426248-48e1717f-1559-4e5a-82a2-12a30e400834.png)

Other settings are identical:
![image](https://user-images.githubusercontent.com/98888664/183426390-76c6fcd8-40bb-40dc-97a0-7f1bd043e109.png)


Q3-1: Why is that ? Shouldn't it be minor?
Q3-2: is this because this is a library-free search? How about library-based searches then?

**Reply by JM-Bader** [reply]
For completeness, please find the logs here: 

library-free search with gene inference
[DA-4O0.4_PL01_DA-SER1.19_libfree_optRes_noLib_GeneInf_.log.txt](https://github.com/vdemichev/DiaNN/files/9290238/DA-4O0.4_PL01_DA-SER1.19_libfree_optRes_noLib_GeneInf_.log.txt)

library-free with Isoform ID inference:
[DA-4O10.00_PL01_DA-SER1.19_Isof.Inf_LibFree_opt.Res_.log.txt](https://github.com/vdemichev/DiaNN/files/9290250/DA-4O10.00_PL01_DA-SER1.19_Isof.Inf_LibFree_opt.Res_.log.txt)

**Reply by vdemichev** [reply]
"It appears that I get drastically fewer quantified proteins in a library-free search when using Protein inference set to Isoform IDs compared to Protein inference set to Genes." - fewer 'unique' proteins because this changes 'proteotypicity' definition. With 'Genes' it counts Genes identified with peptides that are specific to them. With 'isoforms' it counts isoforms identified with peptides that are specific to them. Peptides are very often specific to genes but not to isoforms.

**Reply by JM-Bader** [reply]
I understand that the definition of proteotypicity is changed by the setting. 
However, the resulting behaviour / numbers are unexpected to me. 
It is unexpected to me how the number of protein groups can be higher with gene level protein grouping compared to protein isoform level grouping. 

This example: 
1 Gene has 3 protein isoforms (A,B,C) in the fasta. 
'Low depth' means only razor peptides for the different isoforms are quantified
'high depth' means isoform-specific peptides are quantified

1) Gene level protein grouping + low depth > 1 protein group (GENE) > 1 entry in output table
2) Gene level protein grouping + high depth > 1 protein group (GENE) comprising A,B,C due to coerced 'gene level grouping' >> 1 entry in output table
3) Isoform level protein grouping + low depth > 1 protein group (GENE) > 1 entry in output table
4) Isoform level protein grouping + high depth > 3 protein groups (A,B,C) > 3 entries in output table

So how can it be that when using protein isoform level grouping there are fewer entries than with gene level grouping?

**Reply by vdemichev** [reply]
Where is the number of protein groups higher? This number is never displayed in hte logs.

**Reply by JM-Bader** [reply]
The number is higher in my analysis - loading the main report tsv of the two searches

**Reply by vdemichev** [reply]
How much higher? Actual numbers? Any examples of peptides having different protein groups in each case?

**Reply by vdemichev** [reply]
There's no such thing as gene-level protein grouping, but rather the 'Genes' setting gives a 'hint' to DIA-NN how to group protiens. This is in no way strict. What you might be seeing is that in case there's no gene name associated with any of the sequence IDs matched to a peptide, grouping is just not applied to these...

**Reply by JM-Bader** [reply]
> The number is higher in my analysis - loading the main report tsv of the two searches

Sorry - I was wrong. I usually just rely on the numbers read out from the .tsv files. In this case, the numbers are the same/about the same using either 'gene' or 'isoform' protein grouping. 

the strikingly distinct numbers shown in the log lead me to directly comment here. 
The information in the log have their value - but in cases like this it is very confusing. 
The logs have drastically different numbers (typically 400 for protein isoform-level grouping vs 800 for gene-level grouping) but in the actual tsv report the numbers are comparable....

**Reply by JM-Bader** [reply]
> There's no such thing as gene-level protein grouping, but rather the 'Genes' setting gives a 'hint' to DIA-NN how to group protiens. This is in no way strict. What you might be seeing is that in case there's no gene name associated with any of the sequence IDs matched to a peptide, grouping is just not applied to these...

What does it mean that "Genes setting gives a hint to DIANN how to group proteins" ? So far I assumed that this coerces all protein isoforms deriving from one gene to be in 1 protein group. From what I understand now this is not the case. If not - what is the logic?

**Reply by vdemichev** [reply]
If there are two isoforms, one has a gene name annotated, another - has not, the first will be preferred, for example.

**Reply by JM-Bader** [reply]
> If there are two isoforms, one has a gene name annotated, another - has not, the first will be preferred, for example.

you mean these two isoforms end up in one protein group because only a razor peptide is quantified, the gene annotation of one of them will be used for the entire protein group. Correct? 

So overall, a very mild setting. No coerced grouping at gene level

**Reply by vdemichev** [reply]
Exactly

**Reply by JM-Bader** [reply]
In any case, I suggest the log (also) reports: protein groups for current raw file (not just unique at precursor / protien level).
